realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -974,10 +1101,34 @@ private boolean containsDot(String s) {
     // Sum
 
     /**
+     * Calculate the sum of a given field.
+     *
+     * @param fieldName   The field to sum. Only int, float, and double are supported.
+     * @return            The sum. If the given field has 0 rows or all rows with {@code null} values,
+     *                    {@code 0} will be returned. In calculations of aggregate functions, objects
+     *                    with {@code null} values will be ignored.
+     * @throws            java.lang.IllegalArgumentException if field is not int, float or double.
```

Comment:
`If the field is not a number type.`

Not sure if we should enumerate them all?

`short, int, long, float, double or their boxed counterparts.`